### PR TITLE
Buildserver-oriented memory profiling

### PR DESCRIPTION
**Pull request by aronasorman**
The goal of this patch is to introduce lightweight memory profiling that can be run in the build server/in the background. The data will be used to track how our memory usage grows over time as measured by the benchmarking suite. This isn't designed to help pinpoint where memory leaks happen, see `guppy-heapy` and `memory_profiler` for that.

Oh, and with this change I also introduce a more substantial CircleCI config file.

### CODE REVIEW

**Comment by aronasorman** [comment]
The CircleCI test failures are mostly related to query tests. I hope to remove those soon, and have them measured in a similar way to this (tracked in a CSV rather than hard test failures).

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+machine:
+  python:
+    version: 2.6.8
```
Comment (MCGallaspy) [inline circle.yml]: Why not 2.7.9?

File: kalitectl.py [hunk 2]
```diff
@@ -129,16 +132,16 @@ def __init__(self, status_code):
...
     """
     Takes a list of default arguments and overwrites the defaults with
     contents of updates.
-    
+
     e.g.:
-    
-    udpate_default_args(["--somearg=default"], ["--somearg=overwritten"])
+
+    update_default_args(["--somearg=default"], ["--somearg=overwritten"])
```
Comment (MCGallaspy) [inline kalitectl.py]: udpate is a perfectly cromulent word!

File: circle.yml (hunk 1 above)
Comment (aronasorman) [inline circle.yml]: Unfortunately testing multiple python versions in CircleCI is a bit more involved than Travis, so I went with the lowest version we support for now.

**Comment by MCGallaspy** [comment]
Please add some documentation, regarding intended use and output.

**Comment by aronasorman** [comment]
@MCGallaspy where's the best place to put such documentation? Docstring, dev docs, github wiki?

**Comment by MCGallaspy** [comment]
Probably here: https://github.com/learningequality/ka-lite/tree/develop/sphinx-docs/developer_docs

Be sure to add an entry to `index.rst` as well.

**Comment by aronasorman** [comment]
I should also mention that this will only work for Unix systems.

**Comment by aronasorman** [comment]
@MCGallaspy I created a new file called `profiling.rst`.

**Comment by MCGallaspy** [comment]
Looks good! How seriously should we take failing tests on `circleci` right now?

**Comment by aronasorman** [comment]
Not serious. We should migrate to it once we move away from query tests.

**Pull request merged**